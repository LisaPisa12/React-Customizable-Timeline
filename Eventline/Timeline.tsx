import React from 'react';
import { useSpring, animated, config } from 'react-spring';
import { ThemeProvider } from 'styled-components';
import TimelineItem from './TimelineItem';
import { Timeline, TimelineSection, Title } from './styles';
import { Item } from './Models/Item';
import { Fade } from 'react-awesome-reveal';

type Props = {
    animation?: boolean;
    data?: Item[];
    primaryDarkColor?: string;
    primaryLightColor?: string;
    primaryFont?: string;
    titleShape?: string;
    dotShape?: string;
    lineStyle?: string;
};

const Timeline: React.FC<Props> = ({
    animation = true,
    data,
    primaryDarkColor = '#625261',
    primaryLightColor = '#F5F5DC',
    primaryFont = "'Chivo', sans-serif",
    titleShape = 'circle',
    dotShape = 'circle',
    lineStyle = 'dotted',
}) => {
    const theme = { primaryDarkColor, primaryLightColor, primaryFont, titleShape, dotShape, lineStyle };

    const titleAnimation: Object = useSpring({
        from: {
            transform: `translate3d(0, -4px, 0) scaleY(1.02)`,
        },
        to: {
            transform: 'translate3d(0, 0, 0) scaleY(1.02)',
        },
        config: config.wobbly,
    });

    const timelineAnimation: Object = useSpring({
        from: { transform: 'translate3d(-100%, 0, 0)' },
        to: { transform: 'translate3d(0, 0, 0)' },
        config: { mass: 2, tension: 120, friction: 24 },
        reset: animated,
    } as any);

    return (
        <ThemeProvider theme={theme}>
            <animated.div style={animation ? timelineAnimation : undefined}>
                <Timeline>
                    {data?.map((item, index) => {
                        return (
                            <TimelineSection key={index}>
                                <animated.div style={animation ? titleAnimation : undefined}>
                                    <Title key={item.title}>
                                        <div>{item.title}</div>
                                    </Title>
                                </animated.div>
                                {item.events.map((event, index) => {
                                    return (
                                        <Fade>
                                            <TimelineItem
                                                key={index}
                                                title={event.title}
                                                subtitle={event.subtitle}
                                                content={event.content}
                                                img={event.img}
                                                label={event.label}
                                                location={event.location}
                                                animation={animation}
                                            />
                                        </Fade>
                                    );
                                })}
                            </TimelineSection>
                        );
                    })}
                </Timeline>
            </animated.div>
        </ThemeProvider>
    );
};

export default Timeline;
