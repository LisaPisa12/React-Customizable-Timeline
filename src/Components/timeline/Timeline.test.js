import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Timeline from './Timeline';

test('render correctly', () => {
  const container = document.createElement('div');
  ReactDOM.render(<Timeline />, container);
  expect(container).toMatchSnapshot();
});
