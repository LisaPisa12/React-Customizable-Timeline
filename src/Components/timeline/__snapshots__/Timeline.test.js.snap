// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<div>
  <div
    style="transform: translate3d(-100%, 0, 0);"
  >
    <div
      class="sc-bwzfXH kYoUhp"
    />
  </div>
</div>
`;
